src/processTargets/modifiers/surroundingPair/index.ts

```diff
@@ -33,11 +34,40 @@ export function processSurroundingPair(
   range: Range,
   scopeType: SurroundingPairScopeType
 ): SurroundingPairInfo | null {
-  const document = editor.document;
   const delimiters = complexDelimiterMap[
     scopeType.delimiter as ComplexSurroundingPairName
   ] ?? [scopeType.delimiter];
+  return processSurroundingPairForDelimiters(
+    context,
+    editor,
+    range,
+    scopeType,
+    delimiters
+  );
+}
 
+/**
+ * Applies the surrounding pair modifier to the given selection. First looks to
+ * see if the target is itself adjacent to or contained by a modifier token. If
+ * so it will expand the selection to the opposite delimiter token. Otherwise,
+ * or if the opposite token wasn't found, it will proceed by finding the
+ * smallest pair of delimiters which contains the selection.
+ *
+ * @param context Context to be leveraged by modifier
+ * @param selection The selection to process
+ * @param modifier The surrounding pair modifier information
+ * @param delimiters List of delimiter per names to use
+ * @returns The new selection expanded to the containing surrounding pair or
+ * `null` if none was found
+ */
+export function processSurroundingPairForDelimiters(
+  context: ProcessedTargetsContext,
+  editor: TextEditor,
+  range: Range,
+  scopeType: SurroundingPairScopeType,
+  delimiters: SimpleSurroundingPairName[]
+): SurroundingPairInfo | null {
```

Comment:
I'm also not that happy with it. But it is the same signature as [findSurroundingPairTextBased](https://github.com/cursorless-dev/cursorless/blob/022efb15a6b94ecbef32f7ecbcc6f2b188ac03ef/src/processTargets/modifiers/surroundingPair/findSurroundingPairTextBased.ts#L68-L180) so the problem is deeper. May be just make it so that delimiter can optionally be a list?

Reply:
What's wrong with using a complex scope type, as per link above?

Reply:
Fair point re `findSurroundingPairTextBased`, but I don't think that means the problem is deeper.  We can fix it here using a new complex surrounding pair type  so at least we localise that problem to the impl of surrounding pair

Reply:
I'm fine with that. Just have to decide which solution is the best one.

Reply:
Do you have another solution in mind?

Reply:
Hmm not a huge fan of variadic types like that, plus having it formalised puts its definition in one place, rather than baked into this code

Reply:
I agree that a least always is a better typing. That we are hesitant to do the better solution because of migrations is really indicative in how laborious our migrations are. Ideally it would just be writing a simple migration and be done with it.

Could you elaborate on the complex surrounding pair type and how that would look?